### PR TITLE
Redact Redis and MongoDB credentials from logs (#15)

### LINKED ISSUE #15: Redis and MongoDB credentials are written to application logs
## Summary

Connection configuration formatting includes plaintext credentials, and startup logs emit those formatted values/URLs.

## Evidence

As of `348611b`:

- `RedisConfig::url` embeds username/password and `Display` prints the complete URL (`src/infrastructure/config/redis.rs:19-52`, `82-88`).
- `main` logs `RedisConfig`, and `RedisClient::new` logs the URL again (`src/main.rs:109-112`, `src/infrastructure/redis/client.rs:17-23`).
- `MongoDBConfig::Display` prints the full URI (`src/infrastructure/config/mongo.rs:38-45`).
- `init_mongodb` logs that display value (`src/infrastructure/repositories/mongo_repo.rs:77-82`).
- Both config structs derive `Debug`, which also exposes secret fields.

## Impact

Credentials can leak into local logs, container logs, CI output, observability backends, and incident bundles.

## Proposed direction

Separate connection-string construction from redacted display/debug output. Redact userinfo consistently and avoid logging full URIs.

## Acceptance criteria

- No log/Display/Debug output contains Redis or MongoDB passwords.
- Tests explicitly assert that sentinel secrets never appear.
- Connection errors are sanitized before logging.

## Revalidation

Revalidated on `4b49eca` after the workspace dependency migration; the root cause is still present.

### PR DESCRIPTION
**Pull request by joaquinbejar**
## Summary

Redis and MongoDB credentials were printed in plaintext by the config
`Display`/derived `Debug` impls and logged at three startup sites, leaking
secrets into local logs, container logs, CI output, and observability
backends. This PR redacts credentials from every formatted/logged form while
leaving the actual connection strings untouched.

## Stack

- **Stack:** #15 → #13 → #7 → #10 → #21 → #5 → #4 → #6 → #20 → #19 → #8 → #9 → #11 → #12 → #14 → #17 → #18 → #16 → #22 — **this is PR 1 (bottom of the stack)**
- **Base branch:** `main`
- **Stacked on:** — (first in the chain)
- **Blocks:** the next PR in the stack (#13)

## Changes

- Shared `redact_userinfo` helper (plain string scanning, no new deps) that
  replaces URL userinfo with `***` in any URL-like substring.
- `RedisConfig`: derived `Debug` dropped; manual `Display`/`Debug` render the
  redacted URL. `url()` (the connection value) is unchanged.
- `MongoDBConfig`: derived `Debug`/`Serialize`/`Deserialize` removed (nothing
  in the crate serializes it); manual `Display`/`Debug` redact the URI.
- `RedisClient::new` and `MongoDBClient::new` log redacted forms; driver
  connection errors that can echo the URI are sanitized before logging or
  wrapping into `ChainError`.
- Sentinel tests assert `s3ntinel-pw` never appears in `Display`/`Debug`/
  sanitized-error output; old tests that asserted the leaking output were
  rewritten to expect redaction.

## Testing

- [x] Unit tests added or updated (`LOGLEVEL=WARN cargo test --workspace`, MongoDB up): 229 + 2 passed, 0 failed
- [x] Sentinel-secret assertions on Display / Debug / sanitized errors
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings` clean; `cargo fmt --all --check` clean

## Checklist

- [x] Code follows `rules/global_rules.md` and `CLAUDE.md`
- [x] Errors converge on `ChainError`; no driver error leaks through `pub` signatures
- [x] No `.unwrap()` / `.expect()` / unchecked indexing on request paths
- [x] `tracing` only for logging
- [x] No `unsafe` introduced; no new dependencies
- [x] Connection behavior unchanged (`url()`/`uri` intact)

Known follow-up (out of scope here): `ClickHouseConfig` still derives `Debug`
(its `Display` does not print the password, but `{:?}` would) — same
treatment tracked as a follow-up issue.

Closes #15

### CODE REVIEW

File: src/infrastructure/config/mod.rs [hunk 1]
```diff
@@ -1,3 +1,126 @@
...
+            Some(pos) => {
+                // Byte index just past the "://" scheme separator.
+                let sep = i + pos + 3;
+                result.push_str(&s[i..sep]);
+
+                // The authority section ends at the first '/', whitespace, or
+                // the end of the string.
+                let rest = &s[sep..];
+                let authority_end = rest
+                    .find(|c: char| c == '/' || c.is_whitespace())
```
Comment (joaquinbejar) [inline src/infrastructure/config/mod.rs]: 🔴 **Critical**: this stops scanning before the `@` when a raw credential contains `/` or whitespace. `RedisConfig::url` inserts environment credentials verbatim, so a password such as `p/secret` survives redaction and is written by `Display` and the startup logs. Use a parser with a conservative fallback that redacts through the final userinfo `@`, and add sentinel tests for `/` and whitespace.

**Review by joaquinbejar — COMMENTED** [review]
Credential redaction still exposes configured secrets for delimiter-containing passwords. Intended decision REQUEST_CHANGES, submitted as COMMENT because GitHub does not allow requesting changes on a pull request authored by the current account.

File: src/infrastructure/config/mod.rs (hunk 1 above)
Comment (joaquinbejar) [reply src/infrastructure/config/mod.rs]: Good catch. RedisConfig Display and Debug now build the redacted form directly from the fields so the password never enters a parsed string, MongoDB display goes through a whole-uri redactor that bounds userinfo by the last @ across the entire value, and the mid-text scanner now runs to whitespace and over-redacts a path @ rather than risk a leak. Sentinel tests cover /, whitespace, and @ passwords in 2c16e0d.

**Comment by joaquinbejar** [comment]
Addressed in 2c16e0d, display output is now field-constructed and the scanner is conservative, delimiter sentinel tests added.
